js/__internal/grids/grid_core/column_fixing/m_column_fixing.ts

```diff
@@ -432,41 +434,59 @@ const baseFixedColumns = {
     this.synchronizeRows();
   },
 
-  setColumnWidths(options) {
-    let columns;
-    const visibleColumns = this._columnsController.getVisibleColumns();
+  setColumnWidths(options): void {
     const { widths } = options;
-    const isWidthsSynchronized = widths && widths.length && isDefined(visibleColumns[0].visibleWidth);
-    const { optionNames } = options;
-    const isColumnWidthChanged = optionNames && optionNames.width;
-    let useVisibleColumns = false;
 
-    this.callBase.apply(this, arguments);
+    const visibleColumns = this._columnsController.getVisibleColumns();
+    const isColumnWidthsSynced = widths?.length && visibleColumns.some((column) => isDefined(column.visibleWidth));
+    const isColumnWidthChanged = options.optionNames?.width;
+
+    this.callBase(options);
 
     if (this._fixedTableElement) {
-      const hasAutoWidth = widths && widths.some((width) => width === 'auto');
-      useVisibleColumns = hasAutoWidth && (!isWidthsSynchronized || !this.isScrollbarVisible(true));
+      const hasAutoWidth = widths?.some((width) => width === 'auto');
+      const hasHorizontalScroll = this.isScrollbarVisible(true);
+      const needUseVisibleColumns = hasAutoWidth && (!isColumnWidthsSynced || !hasHorizontalScroll);
 
-      if (useVisibleColumns) {
-        columns = visibleColumns;
+      if (needUseVisibleColumns) {
+        this.setFixedTableColumnWidths(visibleColumns, widths);
+      } else {
+        const fixedColumns = this.getFixedColumns();
+
+        this.setFixedTableColumnWidths(fixedColumns, widths);
```

Comment:
new variable `fixedColumns` is used here, because when `callBase` with `$tableElement =fixedTableElement` and `columns = undefined` was called it [called getColumns inside](https://github.com/DevExpress/DevExtreme/pull/25129/files#diff-a862f0e01d9b654bc096b7d109095ac64c816b242fdb95d8f7e91b4c201a590dL1079) which returned the fixedColumns. So no difference here.